Update INSTALL.md (#378)
pret/pokered: INSTALL.md

apt-get update && apt-get upgrade
```

> WSL has its own file system that's not accessible from Windows, but Windows files *are* accessible from WSL. So you're going to want to install **pokered** within Windows. You'll have to change the **current working directory** every time you open WSL.

Reviewer: >WSL has its own file system that's not accessible from Windows

This statement is incorrect. The WSL file system is indeed accessible from windows by navigating to `\\wsl$\<Distro>\` from a Windows Explorer window.

> So you're going to want to install **pokered** within Windows.

I would argue that we should encourage users to install pokered within the WSL environment, and not windows. WSL, especially WSLv2 is known for poor windows directory access times. If you install pokered inside the WSL environment, there will be dramatically faster build times.

Reviewer: If you are not familiar with the WSL environment enough to rewrite wsl section with this in mind.. then I can make suggestions with changes when I get some time.

Author: > If you are not familiar with the WSL environment enough to rewrite the wsl section with this in mind.. then I can make suggestions with changes when I get some time.

Yes please, I would welcome any improvements you might have in mind. I used Cygwin myself; I have no experience with WSL and can't offer any insight into ways to improve the documentation for using it.

Reviewer: @Mord5 I applied some suggestions; please proofread them first before applying.

> WSL has its own file system that's not accessible from Windows, but Windows files *are* accessible from WSL. So you're going to want to install **pokered** within Windows. You'll have to change the **current working directory** every time you open WSL.

Reviewer: ```suggestion
WSL has its own file system that is accessible by navigating to `\\wsl$\<Distro Name>\` using a Windows Explorer window. Windows files are accessible from WSL by navigating to `/mnt/c/` (equivalent to Windows `C:\`) from within the WSL environment. Due to WSL's poor Windows file access times, especially for WSLv2, it is currently recommended that you download the **pokered** repo to and build from the WSL file system.

(Replace <Distro Name> with the name of the distro you installed into WSL. By default this is Ubuntu. For example: `\\wsl$\Ubuntu\`)
```

Author: ````suggestion
WSL has its own file system that's stored within Windows. You can use Windows Explorer to access the WSL file system by navigating to `\\wsl$\<Distro Name>\`. (Replace <Distro Name> in the example path with the name of the distro you installed into WSL. By default this is Ubuntu. For example: `\\wsl$\Ubuntu\`.)

You can also use WSL to access the Windows file system by navigating to `/mnt/c/` (equivalent to Windows `C:\`) from within the WSL environment. WSL - especially WSLv2 - is slow to access Windows files, so it's recommended that you store **pokered** in WSL's file system and build it from there.

Every time you open WSL, you will have to change the **current working directory** to navigate to wherever it is you have stored **pokered**. For instance, if you want to store **pokered** in your WSL's home directory, enter this command:

```bash
cd /home/<user>/
```

(Replace `<user>` in the example path with your WSL username.)
````
Not sure if the bit about changing working directory is accurate. If you store pokered in `/home/<user>/` do you have to cd every time or is that only if you store somewhere else?

Reviewer: WSL typically launches with the default current directory of your home directory `~` or the equivalent `/home/<user>`. Typically you would want to install pokered in the `/home/<user>/pokered`. So in that instance, yes you would need to cd into `pokered/` after launching WSL.

Author: OK, thank you for clarifying. How do you feel about this text?
````suggestion
WSL has its own file system that's stored within Windows. You can use Windows Explorer to access the WSL file system by navigating to `\\wsl$\<Distro Name>\`. (Replace <Distro Name> in the example path with the name of the distro you installed into WSL. By default this is Ubuntu. For example: `\\wsl$\Ubuntu\`.)

You can also use WSL to access the Windows file system by navigating to `/mnt/c/` (equivalent to Windows `C:\`) from within the WSL environment. WSL - especially WSLv2 - is slow to access Windows files, so it's recommended that you store **pokered** in WSL's file system and build it from there.

The default **current working directory** in WSL is your home directory, `/home/<user>/`. (Replace `<user>` in the example path with your WSL username.) If you store **pokered** in a different directory, you will have to change the **current working directory** every time you open WSL.  For instance, if you want to store **pokered** in `/home/example/` use this command:

```bash
cd /home/example/
```
````

Author: How do you feel about the suggested text?

Reviewer: The example should be for storing inside your home user directory. `/home/<user>/`. 
That way when type type `git clone <pokered-repo-link>` the creates a folder for pokered. `/home/<user>/pokred/`

So rewriting the last sentence should say somethin like. ``For instance, if you want to store **pokered** in your home user directory `/home/<user>/` use this command:``

` ```bash`
` cd /home/<user>/ `
` ``` `

> For example, if you want to store **pokered** in `C:\Users\<user>\Desktop`, enter this command:

Reviewer: ```suggestion
For example, if you want to store **pokered** in your WSL's home directory, enter this command:
```

```bash

> cd /mnt/c/Users/<user>/Desktop

Reviewer: ```suggestion
cd /home/<user>/
```

```

> (The Windows `C:\` drive is called `/mnt/c/` in WSL. Replace `<user>` in the example path with your username.)

Reviewer: ```suggestion
(Replace `<user>` in the example path with your WSL username.)
```

If this works, then follow [the instructions for **Linux**](#linux) below for whatever distribution you installed for WSL.

Otherwise, continue reading below for [the older Windows instructions](#windows-alternate).

## Windows (alternate)

*These instructions are the only option for older versions of Windows, but they also work for Windows 10 or 11 if you prefer not to use the [above](#windows-10-or-11).*

Download [**Cygwin**](http://cygwin.com/install.html): use the installer **setup-x86_64.exe** for 64-bit Windows, **setup-x86.exe** for 32-bit.

Run setup and leave the default settings. At the "**Select Packages**" step, choose to install the following, all of which are in the "**Devel**" category:

- `make`
- `git`
- `gcc-core`

Double click on the text that says "**Skip**" next to each package to select the most recent version to install. Click "**Next**" and go through the remaining installer screens until it's finished.

Once you have finished installing Cygwin, follow the [**rgbds** install instructions](https://rgbds.gbdev.io/install#pre-built) (click **Windows**, then **Cygwin, MSYS2**) to install **rgbds 0.5.2**.

> ***Note** (ignore this if you have never installed **rgbds** before):  

Author: Sure, sounds good.

If you already have an older version of **rgbds**, you will need to update to **0.5.2**. If a version newer than **0.5.2** does not work, try downloading **0.5.2.***

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Author: [StackOverflow ](https://stackoverflow.com/questions/34602584/how-can-i-open-a-cygwin-terminal) agrees with you regarding your choice of terminal, but for new users "open the Cygwin terminal" isn't a self-evident instruction.

If you use mintty as your terminal to launch the bash shell, you have to manually login to bash, whereas cmd via `cygwin.bat` handles that for you. It's a difference of two commands, which is not a lot in the grand scheme of things, but the audience here is specifically users in need of help, so I think it's justifiable to provide instructions that have fewer steps to get to the same place.

Assuming you plan to clone pokered into `home/<user>`, these are the commands needed to build pokered using mintty:

```
/bin/bash --login -i
cd /home/<user>
git clone https://github.com/pret/pokered
cd pokered
make
```

Whereas cmd via `cygwin.bat` only requires the latter 3 commands (which are currently included in the documentation).

Would it be excessive to document both methods of getting there under discussion so that all paths converge on the existing documented commands?

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Author: Have you had a chance to consider the options re: instructing users how to open the cygwin terminal?

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Reviewer 2: It's been a long time since I've done a fresh cygwin install. Does the installer not simply put a mintty shortcut on your desktop?

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Author: It does not, no. (At least, it didn't for me.)

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Reviewer 3: It's an option on the last page of the installer:

![image](https://user-images.githubusercontent.com/8334194/183303033-76aac074-cecf-4f59-afee-2e1e84acfd56.png)

(I'm not sure if it's checked by default and unchecked here since I'm re-running the installer to update packages, or if it's unchecked by default now)

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Author: Does the desktop shortcut go to mintty?

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Reviewer 3: Yes; the shortcut is `C:\cygwin64\bin\mintty.exe -i /Cygwin-Terminal.ico -`.

> Now open the **Cygwin terminal** and enter the following commands. You can open the terminal by double-clicking `Cygwin.bat`, which is in ``C:\cygwin64\``.

Reviewer 4: I should note it's enabled by default, only disabled if there already is one.

> Cygwin has its own file system that's stored within Windows. The Cygwin terminal uses its own syntax, different from Windows, to refer to directories and objects inside the Cygwin file system. In Windows terms, the Cygwin terminal's default **current working directory** is `C:\cygwin64\home\<user>`; in Cygwin terms this same directory is called `/cygdrive/c/cygwin64/home/<user>`. (Replace `<user>` in the example path with your username).

Reviewer 2: That middle sentence is a bit of a mouth full. How about?
```suggestion
Cygwin has its own file system that's stored within Windows. The Cygwin terminal uses a Unix-like file system. In Windows, the Cygwin terminal's default **current working directory** is `C:\cygwin64\home\<user>`; in Cygwin, this same directory is called `/cygdrive/c/cygwin64/home/<user>`. (Replace `<user>` in the example path with your username).
```

Author: How about this:
```suggestion
Cygwin has its own Unix-like file system that's stored within Windows but uses different syntax.
* In Windows, the Cygwin terminal's default **current working directory** is `C:\cygwin64\home\<user>`.
* In Cygwin, this same directory is called `/cygdrive/c/cygwin64/home/<user>`. 
(Replace `<user>` in the example path with your username.)
```
